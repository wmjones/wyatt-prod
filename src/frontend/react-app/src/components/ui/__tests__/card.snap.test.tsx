import React from 'react';
import renderer from 'react-test-renderer';
import {
  Card,
  CardHeader,
  CardFooter,
  CardTitle,
  CardDescription,
  CardContent
} from '../card';

describe('Card Components Snapshots', () => {
  it('renders Card component correctly', () => {
    const tree = renderer
      .create(<Card>Card Content</Card>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders retro Card correctly', () => {
    const tree = renderer
      .create(<Card retro>Retro Card</Card>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders CardHeader component correctly', () => {
    const tree = renderer
      .create(<CardHeader>Header Content</CardHeader>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders CardTitle component correctly', () => {
    const tree = renderer
      .create(<CardTitle>Title Content</CardTitle>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders CardDescription component correctly', () => {
    const tree = renderer
      .create(<CardDescription>Description Content</CardDescription>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders CardContent component correctly', () => {
    const tree = renderer
      .create(<CardContent>Content</CardContent>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders CardFooter component correctly', () => {
    const tree = renderer
      .create(<CardFooter>Footer Content</CardFooter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders complete Card component structure correctly', () => {
    const tree = renderer
      .create(
        <Card>
          <CardHeader>
            <CardTitle>Card Title</CardTitle>
            <CardDescription>Card Description</CardDescription>
          </CardHeader>
          <CardContent>
            Card Content
          </CardContent>
          <CardFooter>
            Card Footer
          </CardFooter>
        </Card>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Card with custom className correctly', () => {
    const tree = renderer
      .create(<Card className="custom-class">Custom Card</Card>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
