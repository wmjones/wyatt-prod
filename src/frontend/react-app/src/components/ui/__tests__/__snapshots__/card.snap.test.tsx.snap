// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Components Snapshots renders Card component correctly 1`] = `
<div
  className="rounded-lg border bg-card text-card-foreground shadow-sm"
>
  Card Content
</div>
`;

exports[`Card Components Snapshots renders Card with custom className correctly 1`] = `
<div
  className="rounded-lg border bg-card text-card-foreground shadow-sm custom-class"
>
  Custom Card
</div>
`;

exports[`Card Components Snapshots renders CardContent component correctly 1`] = `
<div
  className="p-6 pt-0"
>
  Content
</div>
`;

exports[`Card Components Snapshots renders CardDescription component correctly 1`] = `
<p
  className="text-sm text-muted-foreground"
>
  Description Content
</p>
`;

exports[`Card Components Snapshots renders CardFooter component correctly 1`] = `
<div
  className="flex items-center p-6 pt-0"
>
  Footer Content
</div>
`;

exports[`Card Components Snapshots renders CardHeader component correctly 1`] = `
<div
  className="flex flex-col space-y-1.5 p-6"
>
  Header Content
</div>
`;

exports[`Card Components Snapshots renders CardTitle component correctly 1`] = `
<h3
  className="text-2xl font-semibold leading-none tracking-tight"
>
  Title Content
</h3>
`;

exports[`Card Components Snapshots renders complete Card component structure correctly 1`] = `
<div
  className="rounded-lg border bg-card text-card-foreground shadow-sm"
>
  <div
    className="flex flex-col space-y-1.5 p-6"
  >
    <h3
      className="text-2xl font-semibold leading-none tracking-tight"
    >
      Card Title
    </h3>
    <p
      className="text-sm text-muted-foreground"
    >
      Card Description
    </p>
  </div>
  <div
    className="p-6 pt-0"
  >
    Card Content
  </div>
  <div
    className="flex items-center p-6 pt-0"
  >
    Card Footer
  </div>
</div>
`;

exports[`Card Components Snapshots renders retro Card correctly 1`] = `
<div
  className="rounded-lg bg-card text-card-foreground retro-card border-4 border-black shadow-[8px_8px_0px_0px_rgba(0,0,0,1)] transition-all hover:translate-x-[3px] hover:translate-y-[3px] hover:shadow-[5px_5px_0px_0px_rgba(0,0,0,1)]"
>
  Retro Card
</div>
`;
